src/main/java/org/teragrep/JavaPcre.java

```diff
@@ -177,8 +250,9 @@ public void pcre2_jmatch_free(){
     public void pcre2_Jcompile_free(){
         if (re != null){
             LibJavaPcre.INSTANCE.pcre2_jcompile_free(re);
+            LibJavaPcre.INSTANCE.pcre2_ccontext_free(ccontext);
         }else{
-            System.out.print("Error! No data to free.\n");
+            LOGGER.error("No data to free!");
```

Comment:
I think this is worth of an IllegalStateException

Reply:
Added the exception